File: Sorting Algorithms/Bingo Sort/BingoSort.py
def bingo_sort(arr):
    n = len(arr)
    swapped = True

    while swapped:
        swapped = False
        for i in range(0, n-1, 2):
            if arr[i] > arr[i+1]:
                arr[i], arr[i+1] = arr[i+1], arr[i]
                swapped = True

        for i in range(1, n-1, 2):
            if arr[i] > arr[i+1]:
                arr[i], arr[i+1] = arr[i+1], arr[i]
                swapped = True

    return arr

# Example Usage
arr = [64, 34, 25, 12, 22, 11, 90]
sorted_arr = bingo_sort(arr)
print("Sorted array is:", sorted_arr)

File: Sorting Algorithms/Bingo Sort/BingoSort.rb
# Bingo Sort Algorithm

def bingo_sort(arr)
    n = arr.length
    min_idx = 0
  
    (0...n).each do |i|
      min_idx = i
      (i + 1...n).each do |j|
        min_idx = j if arr[j] < arr[min_idx]
      end
      arr[i], arr[min_idx] = arr[min_idx], arr[i]
    end
  
    arr
  end
  
  # Example usage
  arr = [64, 34, 25, 12, 22, 11, 90]
  sorted_arr = bingo_sort(arr)
  puts "Sorted array: #{sorted_arr}"
  